### PR TITLE
remove unnecessary imports

### PR DESCRIPTION
**Pull request by G1Joshi**
Removed all unnecessary imports because all of the used elements are also provided by the other imports.

- This will increase overall pub points.

No version change: 
No functional change for plugin clients

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by G1Joshi** [comment]
I have updated the CHANGELOG!

**Comment by stuartmorgan** [comment]
I just noticed that we still have:
>     # TODO(https://github.com/flutter/flutter/issues/74381):
>    # Clean up existing unnecessary imports, and remove line to ignore.
>    unnecessary_import: ignore

in our analysis_options.yaml. This PR should remove that, since presumably it's fixing all the remaining violations (and if it doesn't currently, it should). That way we are ensuring that this won't regress in the future.

File: packages/camera/camera/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## NEXT
+
+* Remove unnecessary imports.
```
Comment (stuartmorgan) [inline packages/camera/camera/CHANGELOG.md:3]: Nit: This should be "Removes" throughout the PR; see the CHANGELOG style guide linked from the PR checklist.

**Comment by G1Joshi** [comment]
[#74381](https://github.com/flutter/flutter/issues/74381)

**Review by hellohuanlin — APPROVED** [review]
thanks for cleaning this up

**Review by ditman — APPROVED** [review]
This is cool, I can't believe we can remove `dart:async` and `flutter/foundation.dart` from so many places! Good riddance!

File: packages/video_player/video_player/example/lib/main.dart [hunk 2]
```diff
@@ -4,10 +4,6 @@
 
 // ignore_for_file: public_member_api_docs
 
-/// An example of using the plugin, controlling lifecycle and playback of the
```
Comment (asashour) [inline packages/video_player/video_player/example/lib/main.dart:7]: Should this comment be copied to be after the `imports`?

Comment (stuartmorgan) [reply packages/video_player/video_player/example/lib/main.dart:7]: This comment should definitely not be deleted.

Comment (G1Joshi) [reply packages/video_player/video_player/example/lib/main.dart:7]: It was a mistake and fixed now.

**Comment by G1Joshi** [comment]
I have made all the required changes.

**Review by stuartmorgan — APPROVED** [review]
LGTM. Thanks for this cleanup!

**Pull request merged**